### DISCUSSION TITLE
How to enable authentication locally without sharing externally ?

### DISCUSSION THREAD

**Discussion by slapopotamus** [lllyasviel/Fooocus#2021] [Q&A]
I'm looking to enable some basic authentication but I only get the authentication page when I enable "--share" and it also opens an external link. Is there a way to enable authentication for people on the local network without exposing the system externally ?

**Comment by eddyizm** [comment]
It looks like gradio has this feature in the framework [here](https://www.gradio.app/docs/request) but I am not sure if it is implemented anywhere in Fooocus. I will check and get back to you although I have not seen any references in the docs.

**Comment by mashb1t** [comment]
@slapopotamus in https://github.com/lllyasviel/Fooocus/blob/main/webui.py#L621 you should be able to renove the if condition to make it work the way you intend it to use. Haven't checked myself, but seems logical.

**Reply by slapopotamus** [reply]
I'm going to give this a try and provide some feedback :)

**Reply by Schm1tz1** [reply]
Funny - had the same idea. It's not even a one-liner to change it. Theres already an issue and PR reviewed by @mashb1t - needs final review: https://github.com/lllyasviel/Fooocus/pull/2127

**Reply by mashb1t** [reply]
that is the discussion i was referring to 👌
